pkg/server/server.go

```diff
@@ -51,22 +52,30 @@ type stream interface {
 func (s *server) handler(stream stream, selector cache.ResourceSelector) error {
 	// a channel for receiving incoming requests
 	reqCh := make(chan *api.DiscoveryRequest, 0)
+	reqStop := int32(0)
```

Comment:
Hi,

I don't think this is the right way to do it. The reqStop signal will not be consulted until after stream.Recv returns. What I think is needed is some way to close the stream--is there a stream.Close() ?-- that can be called on the way out of this method, so the goroutine will clean itself up.

Reply:
Hi, thanks for taking a look!
I can't seem to find any way to control client stream cancellation in gRPC API. There is no `stream.Close()` as far as I can tell.